src/backend/empty/src/lib.rs

```diff
@@ -97,33 +97,49 @@ impl hal::Device<Backend> for Device {
         unimplemented!()
     }
 
-    fn create_render_pass(&self, _: &[pass::Attachment], _: &[pass::SubpassDesc], _: &[pass::SubpassDependency]) -> () {
+    fn create_render_pass<'a ,T, U, V>(&self, _: T, _: U, _: V) -> ()
+    where
+        T: IntoIterator,
+        T::Item: Borrow<pass::Attachment>,
+        U: IntoIterator,
+        U::Item: Borrow<pass::SubpassDesc<'a>>,
+        V: IntoIterator,
+        V::Item: Borrow<pass::SubpassDependency>,
+    {
         unimplemented!()
     }
 
-    fn create_pipeline_layout(&self, _: &[&()], _: &[(pso::ShaderStageFlags, Range<u32>)]) -> () {
+    fn create_pipeline_layout<'a, T, U>(&self, _: T, _: U) -> ()
+    where
+        T: IntoIterator,
+        T::Item: Borrow<()>,
+        U: IntoIterator,
+        U::Item: Borrow<(pso::ShaderStageFlags, Range<u32>)>,
+    {
         unimplemented!()
     }
 
-    fn create_graphics_pipelines<'a>(
-        &self,
-        _: &[pso::GraphicsPipelineDesc<'a, Backend>],
-    ) -> Vec<Result<(), pso::CreationError>> {
+    fn create_graphics_pipeline<'a, T>(&self, _: T) -> Result<(), pso::CreationError>
```

Comment:
can't we just have `&pso::GraphicsPipelineDesc<'a, Backend>` here?

Reply:
Yes, the generic is completely unnecessary and doesn't provide any benefit.